core/src/apresolve.rs

```diff
@@ -49,27 +57,34 @@ impl ApResolver {
         }
     }
 
-    fn process_data(&self, data: Vec<String>) -> Vec<SocketAddress> {
+    fn process_ap_strings(&self, data: Vec<String>) -> VecDeque<SocketAddress> {
+        let filter_port = self.port_config();
         data.into_iter()
             .filter_map(|ap| {
                 let mut split = ap.rsplitn(2, ':');
                 let port = split.next()?;
-                let host = split.next()?.to_owned();
                 let port: u16 = port.parse().ok()?;
-                if let Some(p) = self.port_config() {
-                    if p != port {
-                        return None;
-                    }
+                let host = split.next()?.to_owned();
+                match filter_port {
+                    Some(filter_port) if filter_port != port => None,
+                    _ => Some((host, port)),
                 }
-                Some((host, port))
             })
             .collect()
     }
 
+    fn parse_resolve_to_access_points(&self, resolve: ApResolveData) -> AccessPoints {
+        AccessPoints {
```

Comment:
You may want to yield `Self` here.

Reply:
Yeah, I wonder why Clippy didn't pick up on this

Reply:
It’s as style preference. I don’t think clippy has a lint for it or at least not by default. But I like DRY.

Reply:
I just took an actual look at the code on my desktop and the function is not actually in `AccessPoints`, but in the `ApResolver` since it uses the `process_ap_strings` function which needs the session reference. So `Self` would be `ApResolver` in this case which wouldn't make much sense